src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1702,10 +1670,45 @@ private void SetNotNullResult(BoundExpression node)
             SetResultType(node, TypeWithState.Create(node.Type, NullableFlowState.NotNull));
         }
 
+        protected override bool IsEmptyStructType(TypeSymbol type)
+        {
+            if (type.TypeKind != TypeKind.Struct)
+            {
+                return false;
+            }
+
+            if (!_emptyStructTypeCache.IsEmptyStructType(type))
+            {
+                return false;
+            }
+
+            if (type.SpecialType != SpecialType.None)
+            {
+                return true;
+            }
+
+            foreach (var member in ((NamedTypeSymbol)type).GetMembersUnordered())
+            {
+                switch (member.Kind)
```

Comment:
>switch (member.Kind) [](start = 16, length = 20)

I am not sure I am following this logic. First of all, it looks order dependent. Depending on what members we get into first, we return different result. And we are not requesting members in any particular order. It would be good to add a comment describing what are we trying to figure out here.

Reply:
Good catch, thanks.

---
In reply to: [284925656](https://github.com/dotnet/roslyn/pull/35765#discussion_r284925656) [](ancestors = 284925656)